components/views/login/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, FormEvent } from 'react'
+import Link from 'next/link'
+
+import { FormProps } from '../../../utilities/types/formTypes'
+import styles from './Login.module.scss'
+import useForm from '../../../utilities/hooks/useForm'
+
+const INITIAL_STATE = {
+    email: '',
+    password: ''
+} as FormProps
+
+const INITIAL_ERROR = {
+    message: 'Enter message',
+    hasError: false
+} as { message: string; hasError: boolean }
+
+const Login = () => {
+    const { input, handleChange, handleReset } = useForm(INITIAL_STATE)
```

Comment:
instead of `input`, this would make more sense reading as `form`. swap the variable name around in both places 😄